### PR TITLE
Support for flashing binary files with openocd nodes

### PR DESCRIPTION
**Pull request by rienafairefr**
sister PRs:
[testbed-webportal #90](https://github.com/iot-lab/testbed-webportal/pull/90)
[testbed-manager #74](https://github.com/iot-lab/testbed-manager/pull/74)
[site-manager #10](https://github.com/iot-lab/site-manager/pull/10)
<-

### CODE REVIEW

File: gateway_code/utils/openocd.py [hunk 1]
```diff
@@ -60,11 +60,16 @@ class OpenOCD(object):
...
+                 ' -c "reset init"'
+                 ' -c "program {0} verify {1}"'
+                 ' -c "shutdown"')
+
     DEBUG = ' -c "reset halt"'
     TIMEOUT = 100
 
     def __init__(self, openocd_args,
-                 verb=False, timeout=TIMEOUT):
+                 verb=True, timeout=TIMEOUT):
```
Comment (aabadie) [inline gateway_code/utils/openocd.py]: Maybe set this back to False ?

File: gateway_code/utils/openocd.py [hunk 2]
```diff
@@ -98,11 +103,14 @@ def reset(self):
         """ Reset """
         return self._call_cmd(self.RESET)
 
-    def flash(self, elf_file):
+    def flash(self, file, binary=False, offset=None):
         """ Flash firmware """
         try:
-            elf_path = common.abspath(elf_file)
-            return self._call_cmd(self.FLASH.format(elf_path))
+            path = common.abspath(file)
```
Comment (aabadie) [inline gateway_code/utils/openocd.py]: `file` is a Python keyword, maybe use `fw_file` instead ?

**Comment by aabadie** [comment]
There are some PEP8 reported by the CI + some unit tests also require an update.

**Comment by rienafairefr** [comment]
@aabadie OK, I've integrated the reviews, and will fix the pep8 and unit tests :-)

**Comment by aabadie** [comment]
FYI, I fixed the remaining failing unit tests and added new ones (there were some new uncovered lines). I went for using return codes != 0 instead of raising exception as it's done this way already.
We need to perform integration tests on related boards before merging this. If it works, it can be merged since it's transparent for the rest of the intrastructure IIUC.

**Comment by aabadie** [comment]
I did more tests of this feature and after updating a few things, I think it can be merged now.

- Integrations tests are passing
- I manually tested the feature on the dev platform (nrf52dk in devsaclay, samr21 in devgrenoble) and it works
- Now edbg flasher also supports the binary mode (and offset). This is required with samr21 boards.

**Comment by aabadie** [comment]
Some more testing on different boards showed that ST MCU ROM is not starting at the address 0 and require to also pass the start address of the ROM. That's what 9e99149 do. I tested it on st-lrwan1 boards and it works. Currently I tested this PR on nrf52dk, nrf52840dk, samr21, arduino-zero and st-lrwan1. And all are working.

**Pull request merged**